tests/contract/test_resource_client.py

```diff
@@ -543,3 +544,21 @@ def test_has_update_handler(resource_client):
     schema = {"handlers": {"update": {"permissions": ["permission"]}}}
     resource_client._update_schema(schema)
     assert resource_client.has_update_handler()
+
+
+def test_assert_create_time(resource_client):
+    resource_client.assert_time(time.time() - 30, time.time(), "CREATE")
```

Comment:
the point of this unit test is to accomplish two things:
* prove that the code you wrote actually does what it intended
* ensure that if the code does change, it does not have unintended consequences.  In other words, if this test starts to fail, either it was on purpose and they need to update the test, or the test failure will show they did something wrong.

I'm not sure these tests accomplish either one.  They don't test every handler, and the timeout value could change a lot without causing this test to fail.

some suggestions:
* you can use `@pytest.mark.parametrize` for this test to do it for CUD actions.
* I would use 59 seconds instead 30.  Do same for the read/list at 29 seconds.  
* parameterize the failure tests at 61/31 seconds.